### ISSUE TITLE
♿ AVT3 ♿ Code Snippet - Screen reader accessibility is poor

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#1531]
User tested with latest version of JAWS 2018, Windows 7, Chrome latest version

JAWS screen reader does not announce a code snippet as code and just reads it as text.

_Note: this could be a larger problem with code blocks and screen readers in general and not specific to our implementation, but it's worth noting here for posterity's sake -- and perhaps work can be done to improve this situation_

**Comment by snidersd** [comment]
@dakahn 
AVT 3 issues found with macOS Mojave, Chrome and VoiceOver.
For the inline code snippet VO does not:
- announce what the button is for. VO should announce that it is a "copy" button.
- announce that the text is a code snippet.
- announce "copied" after the button is pressed the first time. (note: copied is announced after the button is pressed twice).

For the single and multi-line code snippet VO:
- Incorrectly announces that the code snippet is an editable text field.
- Does not announce "copied" after the button is pressed the first time. (note: copied is announced after the button is pressed twice).

**Comment by dakahn** [comment]
Renaming this ticket and we can just use it to track the work on the code snippet more broadly for AVT3 compliance 💃
